src/unix/mod.rs

```diff
@@ -143,6 +143,9 @@ cfg_if! {
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]
         extern {}
+    } else if #[cfg(all(target_vendor = "rumprun", target_os = "netbsd"))] {
+        #[link(name = "m")]
```

Comment:
Do you mean `-nodefaultlibs`? That flag is actually not used in [Rust's Rumprun target](https://github.com/rust-lang/rust/blob/257fa0f479e6a0fc069710a4c2ea52e551bafe97/src/librustc_back/target/x86_64_rumprun_netbsd.rs#L23), since the Rumprun linker always pulls its own libc. So not passing `-lc` explicitly should be fine. 

The reason why I removed the explicit `-lc` is however due to a bug in binutils (as invoked by the Rumprun linker). I haven't really been able to track it down, but for some reason it crashes with an assertion failure if `-lc` is passed to `ld` twice (once explicitly by Rust, and once by Rumprun's linker). 

I can try to get a workaround in the Rumprun linker, so it removes the superfluous `-lc` passed to it, but opted for removing it here, I hope that's okay. Having a redundant `-lc` is not a problem when linking C programs, so I'm not really sure what's going on here.